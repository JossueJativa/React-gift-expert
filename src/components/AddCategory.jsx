import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        const inputVal = inputValue.trim();
        event.preventDefault();
        if (inputVal.length <= 1) return;
        onNewCategory(inputVal);
        setInputValue('');
    }
    return (
        <form onSubmit={ (event) => onSubmit(event) }>
            <input 
                type="text" 
                placeholder="Buscar Gifts" 
                value={ inputValue }
                onChange={ (event) => onInputChange(event)}
            />
        </form>
    )
}