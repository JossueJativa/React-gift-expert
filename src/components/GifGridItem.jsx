import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

// Hacer obligatodio el title y url
GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}