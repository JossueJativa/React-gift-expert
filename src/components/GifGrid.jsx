import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ( { category } ) => {
    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>
            {
                // <LoadingMessage loading={ isLoading } /> Forma con un componente
                // --------------------------------------------------------------
                // isLoading ? <h2>Cargando...</h2> : null Forma con un ternario
                // --------------------------------------------------------------
                isLoading && (<h2>Cargando...</h2>) // Forma con un operador l√≥gico
            }
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifGridItem 
                            key={image.id} 
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}