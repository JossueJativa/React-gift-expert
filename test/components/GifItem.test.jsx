import { render, screen } from '@testing-library/react'; // Import render function
import { GifGridItem } from '../../src/components/GifGridItem';

describe("Propiedad de <GifItem />", () => {
    const gif = {
        title: "Un tÃ­tulo",
        url: "https://localhost/algo.jpg"
    }

    test("Debe hacer match con el snapshoot", () => {
        const { container } = render(<GifGridItem title={gif.title} url={gif.url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y alt de los props', () => {
        render(<GifGridItem title={gif.title} url={gif.url} />);
        // expect(screen.getByRole('img').src).toBe(gif.url);
        // expect(screen.getByRole('img').alt).toBe(gif.title);

        // Otra forma de hacerlo
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(gif.url);
        expect(alt).toBe(gif.title);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifGridItem title={gif.title} url={gif.url} />);
        expect(screen.getByText(gif.title)).toBeTruthy();
    })
});