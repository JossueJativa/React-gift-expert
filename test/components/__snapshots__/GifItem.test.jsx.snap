// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Propiedad de <GifItem /> Debe hacer match con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un título
    </p>
  </div>
</div>
`;
